server/sonar-db-dao/src/test/java/org/sonar/db/component/ComponentDbTester.java

```diff
@@ -59,53 +62,75 @@ private SnapshotDto insertComponentAndSnapshot(ComponentDto component) {
   }
 
   public ComponentDto insertComponent(ComponentDto component) {
-    return insertComponentImpl(component, noExtraConfiguration());
+    return insertComponentImpl(component, null, noExtraConfiguration());
   }
 
-  public ComponentDto insertProject() {
-    return insertComponentImpl(newProjectDto(db.getDefaultOrganization()), noExtraConfiguration());
+  public ComponentDto insertPrivateProject() {
+    return insertComponentImpl(ComponentTesting.newPrivateProjectDto(db.getDefaultOrganization()), true, noExtraConfiguration());
+  }
+
+  public ComponentDto insertPublicProject() {
+    return insertComponentImpl(ComponentTesting.newPublicProjectDto(db.getDefaultOrganization()), false, noExtraConfiguration());
+  }
+
+  @SafeVarargs
+  public final ComponentDto insertPrivateProject(Consumer<ComponentDto>... dtoPopulators) {
+    return insertComponentImpl(ComponentTesting.newPrivateProjectDto(db.getDefaultOrganization()), true, dtoPopulators);
+  }
+
+  @SafeVarargs
+  public final ComponentDto insertPublicProject(Consumer<ComponentDto>... dtoPopulators) {
+    return insertComponentImpl(newPublicProjectDto(db.getDefaultOrganization()), false, dtoPopulators);
   }
 
   @SafeVarargs
-  public final ComponentDto insertProject(Consumer<ComponentDto>... dtoPopulators) {
-    return insertComponentImpl(newProjectDto(db.getDefaultOrganization()), dtoPopulators);
+  public final ComponentDto insertPrivateProject(OrganizationDto organizationDto, Consumer<ComponentDto>... dtoPopulators) {
+    return insertComponentImpl(ComponentTesting.newPrivateProjectDto(organizationDto), true, dtoPopulators);
   }
 
   @SafeVarargs
-  public final ComponentDto insertProject(OrganizationDto organizationDto, Consumer<ComponentDto>... dtoPopulators) {
-    return insertComponentImpl(newProjectDto(organizationDto), dtoPopulators);
+  public final ComponentDto insertPublicProject(OrganizationDto organizationDto, Consumer<ComponentDto>... dtoPopulators) {
+    return insertComponentImpl(newPublicProjectDto(organizationDto), true, dtoPopulators);
```

Comment:
I don't agree:
1. you always have an OrganizationDto available in UTs
2. this is way more code to write. The point of the ComponentDbTester is to make the UT code shorter, less repetitive, not the ComponentDbTester class itself shorter

I'll ignore all similar comments